pkg/credentialprovider/config.go

```diff
@@ -140,19 +154,19 @@ func ReadDockerConfigJSONFile(searchPaths []string) (cfg DockerConfig, err error
 		}
 		cfg, err := readDockerConfigJsonFileFromBytes(contents)
 		if err == nil {
-			glog.V(4).Infof("found .docker/config.json at %s", absDockerConfigFileLocation)
+			glog.V(4).Infof("found %s at %s", dockerConfigJsonFileName, absDockerConfigFileLocation)
 			return cfg, nil
 		}
 	}
-	return nil, fmt.Errorf("couldn't find valid .docker/config.json after checking in %v", searchPaths)
+	return nil, fmt.Errorf("couldn't find valid %s after checking in %v", dockerConfigJsonFileName, searchPaths)
 }
-
 func ReadDockerConfigFile() (cfg DockerConfig, err error) {
-	if cfg, err := ReadDockerConfigJSONFile(nil); err == nil {
-		return cfg, nil
+	if cfg, err = ReadDockerConfigJSONFile(nil); err != nil {
+		if cfg, err = ReadDockercfgFile(nil); err != nil {
+			cfg, err = ReadDockerConfigJsonFileUsedInSecret(nil)
```

Comment:
I would not expect this to look in the default paths for the secret-specific filename